### PR TITLE
Doc Reorg:  Connect Persistent Orgs

### PR DESCRIPTION
**Pull request by Br4nd0R**
Rendered Docs: https://cumulusci.readthedocs.io/en/feature-doc-reorg__connected_orgs/connected_orgs.html

# Changes
* Connect Persistent Orgs Section

### CODE REVIEW

File: docs/connected_orgs.rst [hunk 1]
```diff
@@ -1,14 +1,95 @@
...
+To connect to a persistent org use::
+
+    $ cci org connect <org_name>
+
+This will automatically open a browser window pointed to a Salesforce login page.
+The provided ``<org_name>`` will be the name that CumulusCI associates with org you log into.
+
+
+
+Production and Developer Edition Orgs
```
Comment (davidmreed) [inline docs/connected_orgs.rst:34]: Could we collapse these subsections and just describe when you need the `--sandbox` flag inline?

`--sandbox` is also needed if you're connecting to a scratch org you didn't create.

File: docs/connected_orgs.rst [hunk 2]
```diff
@@ -1,14 +1,95 @@
...
+********************
+For sandboxes, pass the ``--sandbox`` flag along with the org name::
+
+    $ cci org connect <org_name> --sandbox
+
+
+
+Global Orgs
+*******************
+By default, ``cci org connect`` stores the OAuth credentials for connected orgs in a *project specific* keychain.
```
Comment (davidmreed) [inline docs/connected_orgs.rst:52]:
```suggestion
By default, ``cci org connect`` stores the OAuth credentials for connected orgs in a *project-specific* keychain.
```

File: docs/connected_orgs.rst [hunk 3]
```diff
@@ -1,14 +1,95 @@
...
+This means that an org connected while working in Project A's directory, will not be available while working in Project B's directory.
```
Comment (davidmreed) [inline docs/connected_orgs.rst:53]:
```suggestion
This means that an org connected while working in Project A's repository will not be available while working in Project B's repository.
```

File: docs/connected_orgs.rst [hunk 4]
```diff
@@ -1,14 +1,95 @@
...
+
+You can use ``cci org browser`` to ensure that CumulusCI is able to login to the connected org::
+
+    $ cci org browser <org_name>
+
+
+
+Using a Custom Connected App
+----------------------------
+You may want to be ability to configure the connected app used by CumulusCI in a specific way.
```
Comment (davidmreed) [inline docs/connected_orgs.rst:78]: Can we caveat this further? Perhaps

> Most users never need to work with Connected Apps because CumulusCI's out-of-the-box Connected App covers most use cases. If you need to control the Connected App that CumulusCI uses to authenticate for your specific security or compliance requirements, you can create your own Connected App and configure CumulusCI to use  when connecting to orgs.

File: docs/connected_orgs.rst [hunk 5]
```diff
@@ -1,14 +1,95 @@
...
+If this is the case, you can use the ``cci task run connected_app`` command to create the connected app and then manually edit its configuration to your liking.
```
Comment (davidmreed) [inline docs/connected_orgs.rst:79]:
```suggestion
To create a custom Connected App, use the ``connected_app`` task to create the Connected App and then manually edit its configuration to suit your requirements. Make sure to create the Connected App in a persistent org other than a sandbox.

File: docs/connected_orgs.rst [hunk 6]
```diff
@@ -1,14 +1,95 @@
...
+
+You can create a connected app with the label 'cumulusci' and connect it to CumulusCI with::
+
+    $ cci task run connected_app --org <org_name> -o label cumulusci -o connect true
+
+For a full list of options see the `connected_app <TODO>`_ task reference documentation.
+
+After the connected app has been created you can verify that it is connected to CumulusCI by running ``cci service list``.
```
Comment (davidmreed) [inline docs/connected_orgs.rst:87]:
```suggestion
After the Connected App has been created, you can verify that it is connected to CumulusCI by running ``cci service list``.
```

File: docs/connected_orgs.rst [hunk 7]
```diff
@@ -1,14 +1,95 @@
...
+You can edit the connected app's OAuth scopes using the following steps:
```
Comment (davidmreed) [inline docs/connected_orgs.rst:88]:
```suggestion
You can edit the Connected App's OAuth scopes using the following steps:
```

File: docs/connected_orgs.rst [hunk 8]
```diff
@@ -1,14 +1,95 @@
...
+
+#. In lightning, go to Setup --> Apps --> Apps Manager
```
Comment (davidmreed) [inline docs/connected_orgs.rst:90]:
```suggestion
#. In Lightning Experience, go to Setup --> Apps --> Apps Manager
```

File: docs/connected_orgs.rst [hunk 9]
```diff
@@ -1,14 +1,95 @@
...
+#. Click the arrow on the far right side of the row that pertains to the newly created connected app.
```
Comment (davidmreed) [inline docs/connected_orgs.rst:91]:
```suggestion
#. Click the arrow on the far right side of the row that pertains to the newly created Connected App.
```

File: docs/connected_orgs.rst [hunk 10]
```diff
@@ -1,14 +1,95 @@
...
+Connect Persistent Orgs
 =======================
+In order to connect persistent orgs such as a Developer Edition, Enterprise Edition, or Sandbox org to CumulusCI, you first need to have a Connected App configured in a persistent Salesforce org.
 
-Connect a production org
-------------------------
 
-Connect a sandbox org
+
+CumulusCI's Connected App
```
Comment (davidmreed) [inline docs/connected_orgs.rst:7]: Can we avoid starting with Connected Apps and reserve it for the end section? I don't think we want to further the occasional confusion that users _need_ to create Connected Apps.

File: docs/connected_orgs.rst [hunk 11]
```diff
@@ -1,14 +1,95 @@
...
+
+
+Global Orgs
+*******************
+By default, ``cci org connect`` stores the OAuth credentials for connected orgs in a *project specific* keychain.
+This means that an org connected while working in Project A's directory, will not be available while working in Project B's directory.
+
+You can connect an org and make it available to *all* CumulusCI projects on your computer with::
+
+    $ cci org connect <org_name> --global-org
```
Comment (davidmreed) [inline docs/connected_orgs.rst:57]: Is it worth noting that connecting an org does _not_ expose it to SFDX?

**Review by davidmreed — CHANGES_REQUESTED** [review]

File: docs/connected_orgs.rst (hunk 4 above)
Comment (Br4nd0R) [reply docs/connected_orgs.rst:78]: This is a much better intro. Thank you!

File: docs/connected_orgs.rst [hunk 12]
```diff
@@ -1,14 +1,76 @@
-Connect persistent orgs
+Connect Persistent Orgs
 =======================
+In order to connect persistent orgs such as a Developer Edition, Enterprise Edition, or Sandbox org to CumulusCI, you first need to have a Connected App configured in a persistent Salesforce org.
```
Comment (davisagli) [inline docs/connected_orgs.rst:3]: This isn't accurate (in most cases you can just use our default connected app). I'd give this section an intro like this:

In addition to creating scratch orgs [link to scratch org section], you can connect persistent orgs to your CumulusCI project so that you can run tasks and flows on them. This supports use cases such as deploying to a Developer Edition org in order to release a package version, or installing to a sandbox for user acceptance testing.

File: docs/connected_orgs.rst [hunk 13]
```diff
@@ -1,14 +1,76 @@
...
+---------------------------
+To connect to a persistent org use::
 
-Use a custom connected app
+    $ cci org connect <org_name>
+
+This will automatically open a browser window pointed to a Salesforce login page.
+The provided ``<org_name>`` will be the name that CumulusCI associates with org you log into.
+
+If you're connecting to a sandbox use the ``--sandbox`` flag.
```
Comment (davisagli) [inline docs/connected_orgs.rst:16]: I want to call this out in its own section (hence the separate "Connect a sandbox org" header removed in the diff here) to reduce the chance that it gets overlooked.

We should also spell out the different types of orgs for each way (i.e. without the sandbox flag is for production orgs and Developer Edition orgs. With the sandbox flag is for sandboxes and scratch orgs if you want to connect a scratch org that someone else created)

We should also mention the --login-url option.

File: docs/connected_orgs.rst [hunk 14]
```diff
@@ -1,14 +1,76 @@
...
+
+By default, ``cci org connect`` stores the OAuth credentials for connected orgs in a *project-specific* keychain.
```
Comment (davisagli) [inline docs/connected_orgs.rst:18]: I'd suggest putting this paragraph under a "Global Orgs" header, and move it below "Verify Your Connected Orgs". It's a less common use case.

File: docs/connected_orgs.rst [hunk 15]
```diff
@@ -1,14 +1,76 @@
...
+This means that an org connected while working in Project A's repository, will not be available while working in Project B's repository.
+If you would like to make the org available to *all* CumulusCI projects use the ``--global`` flag.
```
Comment (davisagli) [inline docs/connected_orgs.rst:20]: Flag for what? It'd be good to show an example. Also, the flag is actually `--global-org`.

File: docs/connected_orgs.rst [hunk 16]
```diff
@@ -1,14 +1,76 @@
...
+
+.. note::
+    Connecting an org via ``cci org connect`` does *not* expose that org to ``SFDX``.
```
Comment (davisagli) [inline docs/connected_orgs.rst:23]:
```suggestion
    Connecting an org via ``cci org connect`` does *not* expose that org to the Salesforce CLI.
```

(Specifically what we mean is that the org won't show up in `sfdx force:org:list`. It's still possible to run sfdx commands against a connected org, but you have to do so via cci using the `dx` task.)

File: docs/connected_orgs.rst [hunk 17]
```diff
@@ -1,14 +1,76 @@
...
+
+
+Verify Your Connected Orgs
 --------------------------
+You can use ``cci org list`` to look for your org listed under the "Connected Org" table.
+
+The following shows a single persistent org connected to CumulusCI with the name "devhub".
+
+.. image:: images/connected_org.png
```
Comment (davisagli) [inline docs/connected_orgs.rst:32]: I'd suggest showing the output in a code-block unless there's a compelling reason to use an image. Easier to maintain.

File: docs/connected_orgs.rst [hunk 18]
```diff
@@ -1,14 +1,76 @@
...
+
+You can use ``cci org browser`` to ensure that CumulusCI is able to login to the connected org::
+
+    $ cci org browser <org_name>
+
+
+
+CumulusCI's Connected App
+-------------------------
+CumulusCI comes with a Connected App that you can deploy to any org in your ``sfdx`` keychain.
```
Comment (davisagli) [inline docs/connected_orgs.rst:43]: This section looks redundant with the "Using a Custom Connected App" section below, which does a better job of explaining it.

File: docs/connected_orgs.rst [hunk 19]
```diff
@@ -1,14 +1,76 @@
...
+This command also takes care to setup CumulusCI's ``connected_app`` service in the keychain for you.
+If you want to see the information for the Connected App, you can view it with::
+
+    $ cci service info connected_app
+
+
+
+Using a Custom Connected App
+----------------------------
+Most users never need to work with Connected Apps because CumulusCI's out-of-the-box Connected App covers most use cases.
```
Comment (davisagli) [inline docs/connected_orgs.rst:58]:
```suggestion
CumulusCI uses a preconfigured Connected App to authenticate to Salesforce orgs using OAuth2. In most cases this works fine.
```

File: docs/connected_orgs.rst [hunk 20]
```diff
@@ -1,14 +1,76 @@
...
+If you need to control the Connected App that CumulusCI uses to authenticate for your specific security of compliance requirements, you can create your own Connected App and configure CumulusCI to use it when connecting to orgs.
```
Comment (davisagli) [inline docs/connected_orgs.rst:59]:
```suggestion
If you need to control the Connected App that CumulusCI uses to authenticate for your specific security or compliance requirements, you can create your own Connected App and configure CumulusCI to use it when connecting to orgs.
```

File: docs/connected_orgs.rst (hunk 13 above)
Comment (Br4nd0R) [reply docs/connected_orgs.rst:16]: @davidmreed - You had mentioned collapsing these sections previously, do you have issues with going back to a separate subheaders for "Sandboxes"? 

I think that the subheaders also serve a second function as the new docs theme were using uses the section headers in a "contents" section on the right hand side bar. This means that if I'm a user looking to connect a persistent org, I can immediately click on the header that pertains to the type of org I'm looking for.

File: docs/connected_orgs.rst (hunk 18 above)
Comment (Br4nd0R) [reply docs/connected_orgs.rst:43]: @davisagli -  "Using a custom connected app" only mentions in passing the cci has an ootb connected app, and doesn't give steps on how to deploy it. I thought that this was a pre-requisite for even connecting persistent orgs, so wanted ensure there is a header for just deploying the built-in connected app.

Comment (davisagli) [reply docs/connected_orgs.rst:89]: @Br4nd0R "deploying the built-in connected app" is not a thing that anyone has to do. The default connected app exists (in the SFDO Dev Hub) and CumulusCI uses it without needing any configuration. Using the `connected_app` task is ONLY for the edge cases where the user needs a custom connected app.

File: docs/connected_orgs.rst [hunk 21]
```diff
@@ -13,31 +14,76 @@ To connect to a persistent org use::
...
 
 Verify Your Connected Orgs
 --------------------------
 You can use ``cci org list`` to look for your org listed under the "Connected Org" table.
 
 The following shows a single persistent org connected to CumulusCI with the name "devhub".
-
-.. image:: images/connected_org.png
-
+(We are passing the ``--plain`` option for com
```
Comment (davidmreed) [inline docs/connected_orgs.rst:43]: This sentence is cut off.

**Review by davidmreed — CHANGES_REQUESTED** [review]

File: docs/connected_orgs.rst (hunk 13 above)
Comment (davidmreed) [reply docs/connected_orgs.rst]: I think it's fine. I would suggest that we mention that `--sandbox` would also be used for a scratch org you didn't create.

File: docs/connected_orgs.rst [hunk 22]
```diff
@@ -1,14 +1,120 @@
...
+
+.. code-block:: console
+
+    $ cci org browser <org_name>
+
+
+
+Global Orgs
+-----------
+By default, ``cci org connect`` stores the OAuth credentials for connected orgs in a *project specific* keychain. These credentials ensure that an org connected in Project A's directory isn't available when you're working in Project B's directory.
```
Comment (justinvholt) [inline docs/connected_orgs.rst:91]: "These credentials" -- did I phrase that correctly?

File: docs/connected_orgs.rst [hunk 23]
```diff
@@ -1,14 +1,120 @@
...
+
+.. code-block:: console
+
+    $ cci org connect <org_name> --global-org
+
+
+
+Use a Custom Connected App
+----------------------------
+CumulusCI uses a preconfigured Connected App to authenticate to Salesforce orgs that use OAuth2. In most cases this works fine. To control the Connected App for specific security or compliance requirements, create your own Connected App and configure CumulusCI to use it when connecting to orgs.
```
Comment (justinvholt) [inline docs/connected_orgs.rst:103]: "to authenticate to" -- Please confirm this phrasing.

Comment (justinvholt) [inline docs/connected_orgs.rst:103]: "In most cases this works fine." Bit of an odd sentence. What happens when it doesn't work? Do we have a link for troubleshooting?

File: docs/connected_orgs.rst [hunk 24]
```diff
@@ -1,14 +1,133 @@
...
+    │ dev                      │         │ 7    │ X       | dev          │        │
+    ├──────────────────────────┼─────────┼──────┼─────────┼──────────────┼────────┤
+    │ feature                  │         │ 1    │ X       | feature      │        │
+    ├──────────────────────────┼─────────┼──────┼─────────┼──────────────┼────────┤
+    │ prerelease               │         │ 1    │ X       | prerelease   │        │
+    ├──────────────────────────┼─────────┼──────┼─────────┼──────────────┼────────┤
+    │ qa                       │         │ 7    │ X       | qa           │        │
+    ├──────────────────────────┼─────────┼──────┼─────────┼──────────────┼────────┤
+    │ release                  │         │ 1    │ X       | release      │        │
+    └──────────────────────────┴─────────┴──────┴─────────┴──────────────┴────────┘
```
Comment (justinvholt) [inline docs/connected_orgs.rst:76]: I cleaned up the formatting of this table, but I had to get read of the original X's that had a bit more flourish. We can put them back in, but they will mess up the alignment of the table borders and columns.

File: docs/connected_orgs.rst [hunk 25]
```diff
@@ -1,14 +1,133 @@
...
+.. code-block:: console
+
+    $ cci org browser <org_name>
+
+
+
+Global Orgs
+-----------
+
+By default, ``cci org connect`` stores the OAuth credentials for connected orgs in a *project specific* keychain. These credentials ensure that an org connected in Project A's directory isn't accessible when you're working in Project B's directory.
```
Comment (justinvholt) [inline docs/connected_orgs.rst:95]: I changed "available" to "accessible". Just want to confirm that's A-OK.

File: docs/connected_orgs.rst [hunk 26]
```diff
@@ -1,14 +1,133 @@
-Connect persistent orgs
+Connect Persistent Orgs
 =======================
 
-Connect a production org
-------------------------
+In addition to creating `scratch orgs<TODO>`_, you can connect persistent orgs to your CumulusCI project to run tasks and flows on them. This feature supports use cases such as deploying to a Developer Edition org to release a package version, or installing to a sandbox for user acceptance testing.
```
Comment (justinvholt) [inline docs/connected_orgs.rst:4]: So we don't actually say much about what a persistent org is and only touch up a little bit about why they're helpful to use in ``cci``. Does it just mean an org that isn't a DX scratch org?

**Comment by justinvholt** [comment]
@Br4nd0R @davisagli @davidmreed Back in with the flag vs. option discussion. Please keep an eye out for both keywords and let me know if any of them need to be changed.

File: docs/connected_orgs.rst (hunk 22 above)
Comment (davisagli) [reply docs/connected_orgs.rst]: @justinvholt That's a fine way to refer to the credentials (which are typically a refresh token and an access token), but I would rephrase this as "Using a project-specific keychain means that an org connected in Project A's directory isn't available when you're working in Project B's directory." In other words, keeping the two projects' orgs separate is a property of which keychain is used (i.e. where the credentials are stored). "These credentials ensure" make it sound like it's an inherent property of the credentials themselves, which is not the case.

File: docs/connected_orgs.rst (hunk 23 above)
Comment (davisagli) [reply docs/connected_orgs.rst:108]: We meant that there are scenarios where it's not the right solution, not that it breaks intermittently. Examples of those scenarios are:
1. When running CumulusCI in a CI system and using the JWT Oauth2 flow to authenticate to an org (because this flow requires having a private key that was used to sign a certificate that is part of the Connected App configuration, and we can't provide this for the default Connected App)
2. The default Connected App allows any Salesforce user to approve the app, and do anything the user has permission to do. An organization may want to use their own Connected App configuration to enforce additional restrictions on things like who can use it, what they can do, and where they can connect from.

File: docs/connected_orgs.rst (hunk 26 above)
Comment (davisagli) [reply docs/connected_orgs.rst:4]: Yes. This sentence is attempting to clarify that by giving examples of types of persistent orgs (Developer Edition and sandboxes) that the reader is more likely to be familiar with than scratch orgs. A customer's production org is another type of persistent org.

File: docs/connected_orgs.rst (hunk 25 above)
Comment (davidmreed) [reply docs/connected_orgs.rst:95]: Yes. I think there's a different issue with the phrasing, though - "These credentials" doesn't belong as the subject of the second sentence:

```suggestion
By default, ``cci org connect`` stores the OAuth credentials for connected orgs in a *project specific* keychain. This isolated keychain ensures that an org connected in Project A's directory isn't accessible when you're working in Project B's directory.
```

File: docs/connected_orgs.rst [hunk 27]
```diff
@@ -1,14 +1,133 @@
...
+    ├──────────────────────────┼─────────┼──────┼─────────┼──────────────┼────────┤
+    │ qa                       │         │ 7    │ X       | qa           │        │
+    ├──────────────────────────┼─────────┼──────┼─────────┼──────────────┼────────┤
+    │ release                  │         │ 1    │ X       | release      │        │
+    └──────────────────────────┴─────────┴──────┴─────────┴──────────────┴────────┘
+
+    ┌Connected Orgs────┬──────────────────────────────┬────────────┐
+    │ Name   │ Default │ Username                     │ Expires    │
+    ├────────┼─────────┼──────────────────────────────┼────────────┤
+    │ devhub │         │ peter.gibbons@initech.devhub │ Persistent │
```
Comment (davidmreed) [inline docs/connected_orgs.rst:81]: Don't we need to cleanse references to specific media properties from our official docs?

Comment (Br4nd0R) [reply docs/connected_orgs.rst:81]: probably 😢 . Maybe just make it similar: `p.gibbons@inicorp.devhub`?

File: docs/connected_orgs.rst [hunk 28]
```diff
@@ -1,14 +1,133 @@
...
+    └────────┴─────────┴──────────────────────────────┴────────────┘
+
+Make sure that CumulusCI can login to the connected org.
```
Comment (Br4nd0R) [inline docs/connected_orgs.rst:84]:
```suggestion
You can verify a successful connection to the org by logging in with:
```

File: docs/connected_orgs.rst [hunk 29]
```diff
@@ -1,14 +1,133 @@
...
+    $ cci org connect <org_name> --global-org
+
+
+
+Use a Custom Connected App
+----------------------------
+
+CumulusCI uses a preconfigured Connected App to authenticate to Salesforce orgs that use OAuth2. In most cases this works fine. To control the Connected App for specific security or compliance requirements, create your own Connected App and configure CumulusCI to use it when connecting to orgs.
+
+To create a custom Connected App, run the ``connected_app`` task to create the Connected App, and then manually edit its configuration to suit your requirements.
```
Comment (Br4nd0R) [inline docs/connected_orgs.rst:110]:
```suggestion
To create a custom Connected App, run the ``connected_app`` task, and then manually edit its configuration to suit your requirements.
```

File: docs/connected_orgs.rst [hunk 30]
```diff
@@ -1,14 +1,133 @@
...
+
+.. important :: Make sure to create the Connected App in a persistent org other than a sandbox!
+
+    Example: Create a Connected App in the DevHub org connected to ``SFDX`` with the label ``cumulusci`` and set it as the ``connected_app`` service in CumulusCI.
+
+.. code-block:: console
+
+    $ cci task run connected_app -o label cumulusci -o connect true
```
Comment (Br4nd0R) [inline docs/connected_orgs.rst:118]:
```suggestion
    $ cci task run connected_app --label cumulusci --connect true
```

File: docs/connected_orgs.rst [hunk 31]
```diff
@@ -1,14 +1,133 @@
...
+Use a Custom Connected App
+----------------------------
+
+CumulusCI uses a preconfigured Connected App to authenticate to Salesforce orgs that use OAuth2. In most cases this works fine. To control the Connected App for specific security or compliance requirements, create your own Connected App and configure CumulusCI to use it when connecting to orgs.
+
+To create a custom Connected App, run the ``connected_app`` task to create the Connected App, and then manually edit its configuration to suit your requirements.
+
+.. important :: Make sure to create the Connected App in a persistent org other than a sandbox!
+
+    Example: Create a Connected App in the DevHub org connected to ``SFDX`` with the label ``cumulusci`` and set it as the ``connected_app`` service in CumulusCI.
```
Comment (Br4nd0R) [inline docs/connected_orgs.rst:114]: I believe this needs the breaking symbol `..` above the example. Currently the example is rendering as part of the 'important' admonition above.

File: docs/connected_orgs.rst [hunk 32]
```diff
@@ -1,14 +1,133 @@
...
+
+
+Use a Custom Connected App
+----------------------------
+
+CumulusCI uses a preconfigured Connected App to authenticate to Salesforce orgs that use OAuth2. In most cases this works fine. To control the Connected App for specific security or compliance requirements, create your own Connected App and configure CumulusCI to use it when connecting to orgs.
+
+To create a custom Connected App, run the ``connected_app`` task to create the Connected App, and then manually edit its configuration to suit your requirements.
+
+.. important :: Make sure to create the Connected App in a persistent org other than a sandbox!
```
Comment (Br4nd0R) [inline docs/connected_orgs.rst:112]:
```suggestion
.. important :: Make sure to create the Connected App in a production org!
```

File: docs/connected_orgs.rst (hunk 27 above)
Comment (davidmreed) [reply docs/connected_orgs.rst:81]: 🤷‍♂️  I'm not really sure where the line is.

Comment (justinvholt) [reply docs/connected_orgs.rst:81]: We can use my name! I shall live in infamy!

File: docs/connected_orgs.rst (hunk 23 above)
Comment (justinvholt) [reply docs/connected_orgs.rst:108]: @davisagli Is this documented anywhere? It's a really solid explanation.

Comment (davisagli) [reply docs/connected_orgs.rst:108]: @justinvholt We do mention the need for creating a custom connected app in the section about running CumulusCI within a CI system using the JWT oauth2 flow. Feel free to incorporate these examples here to make the explanation clearer, if you think it fits.

File: docs/connected_orgs.rst (hunk 27 above)
Comment (Br4nd0R) [reply docs/connected_orgs.rst]: fine by me :)
